### PR TITLE
Updates to TurboSnap documentation

### PR DESCRIPTION
**Pull request by repomadness**
The majority of this information is from the [TurboSnap logic and docs cleanup](https://www.notion.so/chromatic-ui/TurboSnap-logic-and-docs-cleanup-80b5c2aae1234b64aeee045563acc163?pvs=4) project that started after the Ravio escalation.  The commit related to `turbosnap.mdx` should be almost exactly what was agreed on in the Notion document with @tmeasday and @ethriel3695.  The updates to `setup-turbosnap.mdx` are extra cleanup and giving an uplift to the monorepo section so there's more clarity on what needs to be set for subprojects.  If there's any information that didn't get accurately conveyed, please let me know!

@winkerVSbecks I tried to keep format clean-up minimal on this, but please let me know if you have feedback to share or there were any large misses!

### CODE REVIEW

File: src/content/turbosnap/turbosnap.mdx [hunk 1]
```diff
@@ -15,9 +15,10 @@ TurboSnap is an advanced Chromatic feature that speeds up [UI Tests](/docs/test)
 
 ## How it works
 
-1. TurboSnap considers the Git changes between the current commit and the [ancestor build's](/docs/branching-and-baselines#find-the-ancestor-builds) commit.
-2. Using Webpack's dependency graph, TurboSnap identifies the individual story files affected by those changes.
-3. Chromatic utilizes this information to selectively snapshot the stories defined in those specific story files. It also snapshots any tests that were denied on the ancestor build.
+1. Chromatic determines the [ancestor builds](/docs/branching-and-baselines#find-the-ancestor-builds) associated with your commit.
+2. TurboSnap considers the git changes between the current commit and the ancestor build's commit.  If more than one ancestor build is found, TurboSnap considers the union of the git changes.
+3. Using Webpack's dependency graph, TurboSnap identifies the individual story files affected by those git-identified changes.
```
Comment (tmeasday) [inline src/content/turbosnap/turbosnap.mdx:20]: Webpack or Vite?

File: src/content/turbosnap/turbosnap.mdx [hunk 2]
```diff
@@ -36,14 +37,151 @@ Certain code changes have the potential to impact all stories. To avoid false po
...
+1. Check your `preview.js` file: you may have a global decorator configured that's [imported from an index or barrel file](/docs/setup-turbosnap/#why-changes-are-not-being-detected-correctly), which itself imports a bunch of other files.  This can lead to *all* stories depending on a large swath of seemingly unrelated files.<br /><br />
+Whenever a global Storybook file changes, we run a complete build because global Storybook files can change how *any* story renders.  If any of those files or their imports change, we will trace them as dependent changes.  If the file changes are frequent, you can utilize `--untraced` on these files to help ensure you're not triggering a complete rebuild each time the file changes.
+
+<div class="aside">
+
+Learn more about [how to avoid re-testing for certain files](/docs/setup-turbosnap/#avoid-re-testing-dependent-stories-when-certain-files-changed).
+
+</div>
+
+2. Make sure you have a valid lockfile(s) available and that it's in sync with your `package.json`.  If a lockfile is missing or out ouf sync with `package.json`, we'll have to re-test all stories.
```
Comment (tmeasday) [inline src/content/turbosnap/turbosnap.mdx:52]:
```suggestion
2. Make sure you have a valid lockfile(s) available and that it's in sync with your `package.json`.  If a lockfile is missing or out ouf sync with `package.json`, we'll have to re-test all stories when there are any changes to either file.
```

**Review by tmeasday — APPROVED** [review]
This is great @repomadness

File: src/content/turbosnap/turbosnap.mdx (hunk 1 above)
Comment (repomadness) [reply src/content/turbosnap/turbosnap.mdx:20]: @tmeasday good catch on this one; I'll update the language to be inclusive of both.

File: src/content/turbosnap/turbosnap.mdx [hunk 3]
```diff
@@ -15,9 +15,10 @@ TurboSnap is an advanced Chromatic feature that speeds up [UI Tests](/docs/test)
 
 ## How it works
 
-1. TurboSnap considers the Git changes between the current commit and the [ancestor build's](/docs/branching-and-baselines#find-the-ancestor-builds) commit.
-2. Using Webpack's dependency graph, TurboSnap identifies the individual story files affected by those changes.
-3. Chromatic utilizes this information to selectively snapshot the stories defined in those specific story files. It also snapshots any tests that were denied on the ancestor build.
+1. Chromatic determines the [ancestor builds](/docs/branching-and-baselines#find-the-ancestor-builds) associated with your commit.
+2. TurboSnap considers the git changes between the current commit and the ancestor build's commit.  If more than one ancestor build is found, TurboSnap considers the union of the git changes.
+3. Using the dependency graph generated by Webpack or Vite, TurboSnap identifies the individual story files affected by those git-identified changes.
```
Comment (repomadness) [inline src/content/turbosnap/turbosnap.mdx:20]: @tmeasday let me know if this works or if the language should be specified differently!

**Review by ethriel3695 — APPROVED** [review]
Amazing work @repomadness !!!! 🔥

File: src/content/turbosnap/turbosnap.mdx (hunk 3 above)
Comment (repomadness) [reply src/content/turbosnap/turbosnap.mdx:20]: Gonna go ahead and merge, but if this language needs to be adjusted, let me know, and I'll be happy to open another PR!

**Pull request merged**